src/purchase-client.ts

```diff
@@ -0,0 +1,203 @@
+  onUserPurchase(callback: (transaction: Transaction) => void): void;
+}
+
+export interface Cost {
+  amount: number;
+  type: string;
+}
+
+export interface Product {
+  sku: string;
+  displayName: string;
+  cost: Cost;
+}
+
+export interface Transaction {
+  transactionId: string;
+  product: Product;
+  userId: string;
+  displayName: string;
+  initiator: any;
+  transactionReceipt: string;
+}
+
+// TwitchPurchaseClient implements the basic 'purchase client' interface.
+// This is used by SDK to provide low-level access to twitch bits transactions.
+class TwitchPurchaseClient implements PurchaseClient {
+  /**
+   * purchase will start the twitch bits transaction.
+   *
+   * @since 2.4.0
+   *
+   * @throws {TwitchHelperError} Will throw an error if the Twitch Helper didn't load.
+   * 
+   * @param identifier the twitch sku of the item to be purchased.
+   *
+   * @example
+   * client.purchase("XXSKU000");
+   */
+  public purchase(identifier: string) {
+    if (!window.Twitch.ext.bits) {
+      throw new Error('Twitch helper was not loaded.');
+    }
+
+    window.Twitch?.ext?.bits.useBits(identifier);
+  }
+ 
+  /**
+   * Returns a list of Twitch Bits Products for the current channel.
+   * 
+   * @async
+   * @since 2.4.0
+   * 
+   * @throws {TwitchHelperError} Will throw an error if the Twitch Helper didn't load.
+   *
+   * @return {Promise<[]TwitchBitsProduct>} Resolves with an array of {@link TwitchBitsProduct}
+   * objects for each available sku.
+   *
+   * @example
+   * const products = client.getProducts();
+   */
+  public getProducts(): Promise<TwitchBitsProduct[]> {
+    const target = window.Twitch?.ext?.bits;
+    if (!window.Twitch.ext.bits) {
+      throw new Error('Twitch helper was not loaded.');
+    }
+    
+    return target.getProducts();
+  } 
+
+  /**
+   * onUserPurchase is the interface for adding a post transaction callback.
+   *
+   * @since 2.4.0
+   *
+   * @param callback a function(body)
+   *
+   * @example
+   * client.onUserPurchase(() => {
+   *  console.log("Transaction finished!");
+   * });
+   */
+  public onUserPurchase(callback: (tx: Transaction) => void): void {
+    window.Twitch.ext.bits.onTransactionComplete(
+      (tx: TwitchBitsTransaction) => {
+        if (tx.initiator.toLowerCase() === "current_user") {
+          callback(tx);
+        }
+      }
+    );
+  }
+}
+
+// DevPurchaseClient implements the basic 'purchase client' interface.
+// This is used by SDK to provide low-level access to stubbed transactions.
+class DevPurchaseClient implements PurchaseClient {
+  /**
+   * purchase will start the Dev purchase transaction.
+   *
+   * @since 2.4.0
+   *
+   * @throws {SDKError} Will throw an error if the MuxySDK didn't load.
+   * 
+   * @param identifier the twitch sku of the item to be purchased.
+   *
+   * @example
+   * client.purchase("TESTSKU00");
+   */
+  public purchase(identifier: string) {
+    // TODO: implement some type of purchase for dev testing?
```

Comment:
Loop through all stored callbacks and call each one providing a synthesized `Transaction`.

Reply:
The synthesized transaction should look through `window.MEDKIT_PURCHASABLE_ITEMS` for the proper sku and pull info from that.